### PR TITLE
Feature/assistant

### PR DESCRIPTION
**Pull request by j-konopka**
### Requirements that must be fulfilled:
- [ ] New version in `plugin.json` updated
- [ ] Changelog entry added
- [ ] Plugin can be built (`build-set`)
- [ ] Tested by the author
- [ ] Tested by a second person

[Kanbanize Card](https://plentymarkets.kanbanize.com/ctrl_board/75/cards/159673/details/)
[Kanbanize Card](https://plentymarkets.kanbanize.com/ctrl_board/75/cards/159674/details/)
[Kanbanize Card](https://plentymarkets.kanbanize.com/ctrl_board/75/cards/159676/details/)

@plentymarkets/team-order-payment
____________________
New version must be uploaded from PID 31920.

Further information about the plugin can be found in the [Wiki](https://wiki.plentymarkets.com/display/OP/Payone).

### CODE REVIEW

File: resources/lang/en/MultilingualismConfig.properties [hunk 1]
```diff
@@ -0,0 +1,3 @@
+; sections
+paymentMethod =  "Zahlungsart"
+template =  "Mitteilungen"
```
Comment (MKanitz) [inline resources/lang/en/MultilingualismConfig.properties]: Shouldn't this be in english?

**Review by MKanitz — CHANGES_REQUESTED** [review]
Looks pretty good 👍

File: src/Assistants/PayoneAssistant.php [hunk 2]
```diff
@@ -0,0 +1,625 @@
...
+    /**
+     * @param string $paymentCode
+     * @return array
+     */
+    protected function getDeliveryCountriesConfig(string $paymentCode): array
+    {
+        return [
+            $paymentCode.'AllowedDeliveryCountries' => [
+                'type' => 'checkboxGroup',
+                'defaultValue' => $this->getActiveCountriesValues(),
```
Comment (danielmarx) [inline src/Assistants/PayoneAssistant.php]: Sollte die Liste der activeCountries nicht eingeschränkt werden auf die möglichen Countrys ($this->getDeliveryCountries()). Ist in den meisten Fällen ja nur "DE","AT","CH".

**Review by danielmarx — APPROVED** [review]
Ausser die eine Anmerkung sieht es für mich gut aus.

File: meta/documents/changelog_de.md [hunk 3]
```diff
@@ -1,61 +1,69 @@
 # Release Notes für PAYONE
 
-## 1.3.0 (2020-12-09)
+## 2.0.0
+
+### TODO
+- Es muss der Assistent in jedem verknüpften Set durchlaufen werden um das Plugin zu konfigurieren.
+
+### Hinzugefügt
+- Die Einstellungen für das Payone-Plugin wurden in einen Assistenten überführt. Es ist nun Möglich das Plugin pro Set und pro Mandanten zu konfigurieren.
```
Comment (raphael-adam) [inline meta/documents/changelog_de.md]: `möglich` must be lower case.

File: resources/lang/en/PaymentMethods.properties [hunk 4]
```diff
@@ -0,0 +1,24 @@
+PAYONE_PAYONE_INVOICE=Rechnung
```
Comment (raphael-adam) [inline resources/lang/en/PaymentMethods.properties]: Same as Marvin said, shouldn't this be in english?

File: src/Assistants/PayoneAssistant.php [hunk 5]
```diff
@@ -0,0 +1,637 @@
...
+
+        return $deliveryCountries;
+    }
+
+    /**
+     * Load the active country values
+     */
+    protected function loadActiveCountriesValues()
+    {
+        if ($this->activeCountries === null || $this->activeCountries4AmazonPay) {
```
Comment (MKanitz) [inline src/Assistants/PayoneAssistant.php]: Shouldn't it be ?
```suggestion
        if ($this->activeCountries === null || $this->activeCountries4AmazonPay === null) {
```

**Review by MKanitz — CHANGES_REQUESTED** [review]

**Review by raphael-adam — APPROVED** [review]

File: src/Assistants/PayoneAssistant.php (hunk 5 above)
Comment (j-konopka) [reply src/Assistants/PayoneAssistant.php]: Sure 😄

File: src/Providers/Api/Request/GenericPaymentDataProvider.php [hunk 6]
```diff
@@ -103,19 +103,20 @@ public function getSetOrderReferenceDetailsRequestData(
         string $paymentCode,
         string $workOrderId,
         string $amazonReferenceId,
-        string $currency,
         string $amount,
+        string $currency = null,
```
Comment (MKanitz) [inline src/Providers/Api/Request/GenericPaymentDataProvider.php]: Then you would have to adjust the php doc as well

**Review by MKanitz — CHANGES_REQUESTED** [review]

File: src/Providers/Api/Request/GenericPaymentDataProvider.php (hunk 6 above)
Comment (j-konopka) [reply src/Providers/Api/Request/GenericPaymentDataProvider.php]: Done

**Review by MKanitz — APPROVED** [review]

**Pull request merged**